### PR TITLE
Rename `ISecurityPolicy.authenticated_identity` to `identity`

### LINKED ISSUE #3611: come up with a better name for authenticated_identity
The `authenticated_identity` API is one of the major contributions of the new security policy. However it has already been a source of confusion and I think it is poorly named. For example `request.is_authenticated` is currently testing `request.authenticated_identity is not None` which I'm proposing below to be incorrect. It should instead test `request.authenticated_userid is not None` which remains bw-compat with Pyramid 1.x and frees us to do different things with the `authenticated_identity` object.

The purpose of the API is to allow the security policy to make available to user code an opaque, policy-specific object that they may use with the only contract being that it was created by the security policy. This solves a lot of problems with circular dependencies between things like `unauthenticated_userid`, `request.user`, `authenticated_userid` etc in which a common pattern right now is for a large portion of the "security" work to be done inside `request.user` instead of in the security policy itself. This is great.

The problem is that this opaque object should not necessarily indicate whether a request is authenticated or not. It's important to me that the API is generally useful to indicate to the calling app interesting information about the request such as what principals are included, whether or not the user is logged in, etc. At the same time, I do not want to break the existing API in with `authenticated_userid` returns `None` if the user is not authenticated (it's left to the app / security policy to define what "unauthenticated" actually means).

So I've been trying to come up with a name for this API that does not indicate whether the request is "authenticated" or not. So far, the one I like best is `security_identity` but I wanted to give some other stakeholders a chance to weigh in (@bertjwregeer, @luhn, @merwok, @dstufft come to mind).

If it is more clear the purpose of this object, it can be used much more similarly to how Pyramid 1.x uses `effective_principals` which can be used whether or not a user is logged in. This singular, and admittedly dumb, issue is the reason I haven't shipped 2.0 yet and naming is hard and so I've just dragged my feet on it. I'm very much leaning toward `security_identity`, or `security_data`, or something to that effect. I want it to potentially contain access token information / metadata regardless of authenticated status of the request.

### PR DESCRIPTION
**Pull request by luhn**
Fixes #3611

### CODE REVIEW

**Review by bertjwregeer — COMMENTED** [review]
This is great, thank you @luhn! Just a small change to the wording for the documentation of `identity`, maybe have @stevepiercy take a look at it too.

**Review by stevepiercy — APPROVED** [review]

File: src/pyramid/interfaces.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def app_iter_range(start, stop):
         """Return a new app_iter built from the response app_iter that
         serves up only the given start:stop range."""
 
-    authenticated_identity = Attribute(
+    identity = Attribute(
         """An object representing the authenticated user, as determined by
         the security policy in use, or ``None`` for unauthenticated requests.
         The object's class and meaning is defined by the security policy."""
```
[the remark this answers is not in the document]
Comment (stevepiercy) [reply src/pyramid/interfaces.py:119]: If I understand the attribute correctly, I'd suggest the following, plucking a bit from `class ISecurityPolicy(Interface)`.

```
""" An object representing the current request of the user, as determined by the security policy in use. The object's class and meaning is defined by the security policy. """
```

I would ultimately defer to @mmerickel on the most accurate and descriptive wording.

[the remark this answers is not in the document]
Comment (mmerickel) [reply src/pyramid/interfaces.py:119]: > An object containing authentication information related to the current request. The object's type and meaning is defined by the configured security policy.

**Review by mmerickel — CHANGES_REQUESTED** [review]

**Comment by luhn** [comment]
Updated docstring as requested.

**Review by mmerickel — APPROVED** [review]

**Comment by mmerickel** [comment]
thanks @luhn!

**Comment by bertjwregeer** [comment]
\o/

**Pull request merged**